Distance Sensor Plugin Prerequisites

xabierolaz opened the pull request:
All prerequisites has been made to have the distance sensor plugin working,  before getting into the actual SDF sensor

### Files created:

- **Distance.proto** 
- **gazebo_distance_plugin.cpp** 
- **gazebo_distance_plugin.h** 

### Files updated:
- **Cmakelists.txt**

### **Cmake**  output

~/workspace/aircraft-plugins-new/build$ cmake ..
```
-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   system
--   thread
--   timer
--   chrono
--   date_time
--   atomic
-- Found DART: /usr/local/include (Required is at least version &quot;6.6&quot;) found components:  dart 
-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   thread
--   system
--   filesystem
--   program_options
--   regex
--   iostreams
--   date_time
--   chrono
--   atomic
-- Found Protobuf: /usr/lib/x86_64-linux-gnu/libprotobuf.so;-lpthread (found version &quot;3.0.0&quot;) 
-- Boost version: 1.65.1
-- Looking for OGRE...
-- Found Ogre Ghadamon (1.9.0)
-- Found OGRE: optimized;/usr/lib/x86_64-linux-gnu/libOgreMain.so;debug;/usr/lib/x86_64-linux-gnu/libOgreMain.so
-- Looking for OGRE_Paging...
-- Found OGRE_Paging: optimized;/usr/lib/x86_64-linux-gnu/libOgrePaging.so;debug;/usr/lib/x86_64-linux-gnu/libOgrePaging.so
-- Looking for OGRE_Terrain...
-- Found OGRE_Terrain: optimized;/usr/lib/x86_64-linux-gnu/libOgreTerrain.so;debug;/usr/lib/x86_64-linux-gnu/libOgreTerrain.so
-- Looking for OGRE_Property...
-- Found OGRE_Property: optimized;/usr/lib/x86_64-linux-gnu/libOgreProperty.so;debug;/usr/lib/x86_64-linux-gnu/libOgreProperty.so
-- Looking for OGRE_RTShaderSystem...
-- Found OGRE_RTShaderSystem: optimized;/usr/lib/x86_64-linux-gnu/libOgreRTShaderSystem.so;debug;/usr/lib/x86_64-linux-gnu/libOgreRTShaderSystem.so
-- Looking for OGRE_Volume...
-- Found OGRE_Volume: optimized;/usr/lib/x86_64-linux-gnu/libOgreVolume.so;debug;/usr/lib/x86_64-linux-gnu/libOgreVolume.so
-- Looking for OGRE_Overlay...
-- Found OGRE_Overlay: optimized;/usr/lib/x86_64-linux-gnu/libOgreOverlay.so;debug;/usr/lib/x86_64-linux-gnu/libOgreOverlay.so
-- Found Protobuf: /usr/lib/x86_64-linux-gnu/libprotobuf.so;-lpthread;-lpthread (found suitable version &quot;3.0.0&quot;, minimum required is &quot;2.3.0&quot;) 
-- Config-file not installed for ZeroMQ -- checking for pkg-config
-- Checking for module &apos;libzmq &gt;= 4&apos;
--   Found libzmq , version 4.2.5
-- Checking for module &apos;uuid&apos;
--   Found uuid, version 2.31.1
-- Checking for module &apos;tinyxml2&apos;
--   Found tinyxml2, version 6.0.0
-- Looking for dlfcn.h - found
-- Looking for libdl - found
-- FreeImage.pc not found, we will search for FreeImage_INCLUDE_DIRS and FreeImage_LIBRARIES
-- Checking for module &apos;gts&apos;
--   Found gts, version 0.7.6
-- Checking for module &apos;libswscale&apos;
--   Found libswscale, version 4.8.100
-- Checking for module &apos;libavdevice &gt;= 56.4.100&apos;
--   Found libavdevice , version 57.10.100
-- Checking for module &apos;libavformat&apos;
--   Found libavformat, version 57.83.100
-- Checking for module &apos;libavcodec&apos;
--   Found libavcodec, version 57.107.100
-- Checking for module &apos;libavutil&apos;
--   Found libavutil, version 55.78.100
-- Checking for module &apos;jsoncpp&apos;
--   Found jsoncpp, version 1.7.4
-- Checking for module &apos;yaml-0.1&apos;
--   Found yaml-0.1, version 0.1.7
-- Checking for module &apos;libzip&apos;
--   Found libzip, version 1.1.2
-- Found Protobuf: /usr/lib/x86_64-linux-gnu/libprotobuf.so;-lpthread;-lpthread;-lpthread (found version &quot;3.0.0&quot;) 
Gazebo version: 10.1
-- Configuring done
-- Generating done
-- Build files have been written to: /home/cuda/workspace/aircraft-plugins-new/build
```
### **make** output

cuda@cuda~/workspace/aircraft-plugins-new/build$ **make**
```
[ 62%] Built target sensor_msgs
[ 75%] Built target control_msgs
[ 83%] Built target gazebo_motor_model
[ 91%] Built target gazebo_imu_plugin
[100%] Built target gazebo_distance_plugin
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)
<!-- Reviewable:end -->

xabierolaz committed:
Updated Cmakelists to include Distance Sensor

xabierolaz committed:
First Commit

xabierolaz committed:
Update README.md

xabierolaz committed:
Update README.md

xabierolaz committed:
Update README.md

xabierolaz committed:
last_commit_no_errors

wil3 reviewed (changes requested):
Mostly needs code clean up but looks pretty good otherwise

> Reviewed 1 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-:-MSAy0h96CXh8PW5I7EG:b-1n0bax)* status: 1 of 4 files reviewed, 14 unresolved discussions (waiting on @xabierolaz)

___
*[msgs/Distance.proto, line 3 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAv5K49aOxLP-6QiYu:-MSAv5K49aOxLP-6QiYv:b-y7l872) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/msgs/Distance.proto#L3)):*
> ```protobuf
> syntax = "proto2";
> package sensor_msgs.msgs;
> import "Float.proto";
> ```

This import isn't needed
___
*[msgs/Distance.proto, line 8 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAvtdC1AQJNo5qUOWj:-MSAvtdC1AQJNo5qUOWk:b-k3uruy) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/msgs/Distance.proto#L8)):*
> ```protobuf
> {
>    required float ground_distance   = 1;
>    required int64 time_usec = 2;
> ```

Not needed, the message State.proto in gymfc has the sim time
___
*[src/gazebo_distance_plugin.h, line 37 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAw4_SCY-XSnaYRmYA:-MSAw4_SCY-XSnaYRmYB:b-uv7i6k) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.h#L37)):*
> ```objc
> namespace ground_distance_units{
>     static const std::string METERS = "m" ;
>     
> ```

remove whitespace
___
*[src/gazebo_distance_plugin.h, line 44 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAwDTk9HFj8Jpq_5UM:-MSAwDTk9HFj8Jpq_5UN:b-x50k7x) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.h#L44)):*
> ```objc
> // Default values for use with ADIS16448 IMU
> static constexpr double kDefaultAdisMaxRange =
>     20.0;
> ```

Keep on same line
___
*[src/gazebo_distance_plugin.h, line 102 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAwOIG7cWMugUt1nkZ:-MSAwOIG7cWMugUt1nk_:b-uv7i6k) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.h#L102)):*
> ```objc
>   sensor_msgs::msgs::Distance distance_message_;
> 
>   
> ```

remove whitespace
___
*[src/gazebo_distance_plugin.cpp, line 22 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAx51k0URs21YVuMZ7:-MSAx51k0URs21YVuMZ8:b-l6mwxj) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L22)):*
> ```cpp
> 
> #include "gazebo_distance_plugin.h"
> //#include "gazebo_imu_plugin.h"
> ```

remove
___
*[src/gazebo_distance_plugin.cpp, line 84 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxCky1XEKUJCkE_E4:-MSAxCky1XEKUJCkE_E5:b-85983a) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L84)):*
> ```cpp
>   frame_id_ = link_name_;
> 
>   getSdfParam<std::string>(_sdf, "distanceTopic", distance_topic_, kDefaultDistanceTopic);
> ```

you load the topic here, why dont you use it below?
___
*[src/gazebo_distance_plugin.cpp, line 86 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxJx3-a_vFIeSXjLu:-MSAxJx3-a_vFIeSXjLv:b-sj7pgj) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L86)):*
> ```cpp
>   getSdfParam<std::string>(_sdf, "distanceTopic", distance_topic_, kDefaultDistanceTopic);
>   
>   getSdfParam<double>(_sdf, "MaxRange",
> ```

keep consistent with camelcase MaxRange -> maxRange
___
*[src/gazebo_distance_plugin.cpp, line 90 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxWDBEoTUQ3MEoWfj:-MSAxWDC5eWzqHShEpLt:bvxxn3x) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L90)):*
> ```
> ground_distance_units
> ```
does this ever get used?
___
*[src/gazebo_distance_plugin.cpp, line 111 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAws46DJkDfYf5IuXE:-MSAws46DJkDfYf5IuXF:bmcmwhj) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L111)):*
> ```cpp
> 
>   // Specify queue limit and rate, make this configurable
>   distance_pub_ = node_handle_->Advertise<sensor_msgs::msgs::Distance>("/aircraft/sensor/distance");
> ```

As comment says, this needs to be configurable, look at gazebo_motor_model.cpp for an example
___
*[src/gazebo_distance_plugin.cpp, line 119 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAw_l155-rtbo3BhUb:-MSAw_l155-rtbo3BhUc:b-y8aqqo) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L119)):*
> ```cpp
> 
> 
>   // Fill imu message.
> ```

remove unused code
___
*[src/gazebo_distance_plugin.cpp, line 147 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAwWII54totUMOk31o:-MSAwWII54totUMOk31p:b-uv7i6k) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L147)):*
> ```cpp
> 
>   ignition::math::Vector3d position = T_W_I.Pos();
>   
> ```

remove whitespace
___
*[src/gazebo_distance_plugin.cpp, line 150 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAwYu--xIWzOqshYl8:-MSAwYu--xIWzOqshYl9:b-y8aqqo) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L150)):*
> ```cpp
>   float height = position.Z();
> 
>   // Fill IMU message.
> ```

remove unused code
___
*[src/gazebo_distance_plugin.cpp, line 160 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxj3oC_JLjhLfgxG3:-MSAxj3oC_JLjhLfgxG4:b-l6mwxj) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L160)):*
> ```cpp
>   // imu_message_.orientation.y = 0;
>   // imu_message_.orientation.z = 0;
>   distance_message_.set_time_usec(_info.simTime.sec * 1000000 + _info.simTime.nsec / 1000);
> ```

remove


<!-- Sent from Reviewable.io -->

xabierolaz commented:
Hi @wil3, changed most of things as you requested, Just some few minor things, as this 

msgs/Distance.proto, line 8 at r1 (raw file):
{
   required float ground_distance   = 1;
   **required int64 time_usec = 2;**     //   about this line not being needed (as you say, State.proto in gymfc has the sim time)
I have updated and pulled again with all the changes.
**if I remove it, then we get one compilation error:**

```
cuda@cuda:~/workspace/aircraft-plugins-new/build$ cmake ..
-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   system
--   thread
--   timer
--   chrono
--   date_time
--   atomic
-- Found DART: /usr/local/include (Required is at least version "6.6") found components:  dart 
-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   thread
--   system
--   filesystem
--   program_options
--   regex
--   iostreams
--   date_time
--   chrono
--   atomic
-- Found Protobuf: /usr/lib/x86_64-linux-gnu/libprotobuf.so;-lpthread (found version "3.0.0") 
-- Boost version: 1.65.1
-- Looking for OGRE...
-- Found Ogre Ghadamon (1.9.0)
-- Found OGRE: optimized;/usr/lib/x86_64-linux-gnu/libOgreMain.so;debug;/usr/lib/x86_64-linux-gnu/libOgreMain.so
-- Looking for OGRE_Paging...
-- Found OGRE_Paging: optimized;/usr/lib/x86_64-linux-gnu/libOgrePaging.so;debug;/usr/lib/x86_64-linux-gnu/libOgrePaging.so
-- Looking for OGRE_Terrain...
-- Found OGRE_Terrain: optimized;/usr/lib/x86_64-linux-gnu/libOgreTerrain.so;debug;/usr/lib/x86_64-linux-gnu/libOgreTerrain.so
-- Looking for OGRE_Property...
-- Found OGRE_Property: optimized;/usr/lib/x86_64-linux-gnu/libOgreProperty.so;debug;/usr/lib/x86_64-linux-gnu/libOgreProperty.so
-- Looking for OGRE_RTShaderSystem...
-- Found OGRE_RTShaderSystem: optimized;/usr/lib/x86_64-linux-gnu/libOgreRTShaderSystem.so;debug;/usr/lib/x86_64-linux-gnu/libOgreRTShaderSystem.so
-- Looking for OGRE_Volume...
-- Found OGRE_Volume: optimized;/usr/lib/x86_64-linux-gnu/libOgreVolume.so;debug;/usr/lib/x86_64-linux-gnu/libOgreVolume.so
-- Looking for OGRE_Overlay...
-- Found OGRE_Overlay: optimized;/usr/lib/x86_64-linux-gnu/libOgreOverlay.so;debug;/usr/lib/x86_64-linux-gnu/libOgreOverlay.so
-- Found Protobuf: /usr/lib/x86_64-linux-gnu/libprotobuf.so;-lpthread;-lpthread (found suitable version "3.0.0", minimum required is "2.3.0") 
-- Config-file not installed for ZeroMQ -- checking for pkg-config
-- Checking for module 'libzmq >= 4'
--   Found libzmq , version 4.2.5
-- Checking for module 'uuid'
--   Found uuid, version 2.31.1
-- Checking for module 'tinyxml2'
--   Found tinyxml2, version 6.0.0
-- Looking for dlfcn.h - found
-- Looking for libdl - found
-- FreeImage.pc not found, we will search for FreeImage_INCLUDE_DIRS and FreeImage_LIBRARIES
-- Checking for module 'gts'
--   Found gts, version 0.7.6
-- Checking for module 'libswscale'
--   Found libswscale, version 4.8.100
-- Checking for module 'libavdevice >= 56.4.100'
--   Found libavdevice , version 57.10.100
-- Checking for module 'libavformat'
--   Found libavformat, version 57.83.100
-- Checking for module 'libavcodec'
--   Found libavcodec, version 57.107.100
-- Checking for module 'libavutil'
--   Found libavutil, version 55.78.100
-- Checking for module 'jsoncpp'
--   Found jsoncpp, version 1.7.4
-- Checking for module 'yaml-0.1'
--   Found yaml-0.1, version 0.1.7
-- Checking for module 'libzip'
--   Found libzip, version 1.1.2
-- Found Protobuf: /usr/lib/x86_64-linux-gnu/libprotobuf.so;-lpthread;-lpthread;-lpthread (found version "3.0.0") 
Gazebo version: 10.1
-- Configuring done
-- Generating done
-- Build files have been written to: /home/cuda/workspace/aircraft-plugins-new/build
```
```
cuda@cuda:~/workspace/aircraft-plugins-new/build$ make
[  4%] Running C++ protocol buffer compiler on msgs/Distance.proto
Scanning dependencies of target sensor_msgs
[  8%] Building CXX object CMakeFiles/sensor_msgs.dir/Distance.pb.cc.o
[ 12%] Linking CXX shared library libsensor_msgs.so
[ 62%] Built target sensor_msgs
[ 75%] Built target control_msgs
[ 79%] Linking CXX shared library libgazebo_motor_model.so
[ 83%] Built target gazebo_motor_model
[ 87%] Linking CXX shared library libgazebo_imu_plugin.so
[ 91%] Built target gazebo_imu_plugin
Scanning dependencies of target gazebo_distance_plugin
[ 95%] Building CXX object CMakeFiles/gazebo_distance_plugin.dir/src/gazebo_distance_plugin.cpp.o
/home/cuda/workspace/aircraft-plugins-new/src/gazebo_distance_plugin.cpp: In member function ‘void gazebo::GazeboDistancePlugin::OnUpdate(const gazebo::common::UpdateInfo&)’:
/home/cuda/workspace/aircraft-plugins-new/src/gazebo_distance_plugin.cpp:132:21: error: ‘class sensor_msgs::msgs::Distance’ has no member named ‘set_time_usec’
   distance_message_.set_time_usec(_info.simTime.sec * 1000000 + _info.simTime.nsec / 1000);
                     ^~~~~~~~~~~~~
CMakeFiles/gazebo_distance_plugin.dir/build.make:62: recipe for target 'CMakeFiles/gazebo_distance_plugin.dir/src/gazebo_distance_plugin.cpp.o' failed
make[2]: *** [CMakeFiles/gazebo_distance_plugin.dir/src/gazebo_distance_plugin.cpp.o] Error 1
CMakeFiles/Makefile2:218: recipe for target 'CMakeFiles/gazebo_distance_plugin.dir/all' failed
make[1]: *** [CMakeFiles/gazebo_distance_plugin.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

xabierolaz committed:
Wil's feedback changes

xabierolaz committed:
Merge branch 'master' of https://github.com/xabierolaz/gymfc-aircraft-plugins

xabierolaz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: 1 of 6 files reviewed, 14 unresolved discussions (waiting on @wil3 and @xabierolaz)

___
*[msgs/Distance.proto, line 3 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAv5K49aOxLP-6QiYu:-MSCRBIj4tgPFIkr5eGS:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/msgs/Distance.proto#L3)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

This import isn't needed
</blockquote></details>

Done.
___
*[msgs/Distance.proto, line 8 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAvtdC1AQJNo5qUOWj:-MSCVJ3TCM9GsXn4MU44:b89ura) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/msgs/Distance.proto#L8)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

Not needed, the message State.proto in gymfc has the sim time
</blockquote></details>

Understand the issue, but It actually wont compile without that line
___
*[src/gazebo_distance_plugin.h, line 37 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAw4_SCY-XSnaYRmYA:-MSCREVvDUxlV-UO5S6X:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.h#L37)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

remove whitespace
</blockquote></details>

Done.
___
*[src/gazebo_distance_plugin.h, line 44 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAwDTk9HFj8Jpq_5UM:-MSCRFOQDcfTGlp8nDXP:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.h#L44)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

Keep on same line
</blockquote></details>

Done.
___
*[src/gazebo_distance_plugin.h, line 102 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAwOIG7cWMugUt1nkZ:-MSCRGRV2OaA-Mfs58nX:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.h#L102)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

remove whitespace
</blockquote></details>

Done.
___
*[src/gazebo_distance_plugin.cpp, line 22 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAx51k0URs21YVuMZ7:-MSCRISu29KhmP2kZbDz:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L22)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

remove
</blockquote></details>

Done.
___
*[src/gazebo_distance_plugin.cpp, line 84 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxCky1XEKUJCkE_E4:-MSChc-rAe6A2lunD9iC:bmvy7xq) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L84)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

you load the topic here, why dont you use it below?
</blockquote></details>

Can you elaborate on this, please
___
*[src/gazebo_distance_plugin.cpp, line 86 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxJx3-a_vFIeSXjLu:-MSCV2Z12fmdWxUIu2oM:b-ylz3ax) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L86)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

keep consistent with camelcase MaxRange -> maxRange
</blockquote></details>

Did not find any issue regarding this in.cpp or .h , could you elaborate this ?
___
*[src/gazebo_distance_plugin.cpp, line 90 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxWDBEoTUQ3MEoWfj:-MSCU--fAiu9bZq3XgMD:b49msps) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L90)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

> ```
> ground_distance_units
> ```
does this ever get used?
</blockquote></details>

It was not, removed
___
*[src/gazebo_distance_plugin.cpp, line 111 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAws46DJkDfYf5IuXE:-MSCTOmWCRzCy73X0d1h:bpm28kg) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L111)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

As comment says, this needs to be configurable, look at gazebo_motor_model.cpp for an example
</blockquote></details>

Done, following motor model as you suggested
___
*[src/gazebo_distance_plugin.cpp, line 119 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAw_l155-rtbo3BhUb:-MSCi8qO0lZ81DzrbL8r:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L119)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

remove unused code
</blockquote></details>

Done.
___
*[src/gazebo_distance_plugin.cpp, line 150 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAwYu--xIWzOqshYl8:-MSCS35x1V_iglgdmQzd:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L150)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

remove unused code
</blockquote></details>

Done.
___
*[src/gazebo_distance_plugin.cpp, line 160 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxj3oC_JLjhLfgxG3:-MSCi9xIDgTHeF0YsGrA:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L160)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

remove
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

xabierolaz reviewed (commented):
> Reviewed 4 of 4 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 14 unresolved discussions (waiting on @wil3 and @xabierolaz)



<!-- Sent from Reviewable.io -->

xabierolaz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 14 unresolved discussions (waiting on @wil3)

___
*[src/gazebo_distance_plugin.cpp, line 147 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAwWII54totUMOk31o:-MSCmIc69sHemir0BOtW:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L147)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

remove whitespace
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

wil3 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 18 unresolved discussions (waiting on @wil3 and @xabierolaz)

*[a discussion](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSPcY3q6VTBR11IBvau:-MSPcY3q6VTBR11IBvav:bvsd17w) (no related file):*
Below you've said you've made  changes but they haven't been made. I double checked some on your branch and they don't exist there either. Before responding they are done make sure the current revision actually reflects the change being made. 

---
___
*[.DS_Store, line 0 at r2](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSPc0w8C7QERmnHQ0sp:-MSPc0w8C7QERmnHQ0sq:bc9ix63) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/1642ce295c91e7cb9b5df1f2c8a61ac34978e465/.DS_Store#L0)):*
what is this file? delete
___
*[Icon
, line 0 at r2](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSPc3Pa1j4itGcSQzQ3:-MSPc3Pa1j4itGcSQzQ4:b-rtagwg) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/60551f5eff70d68923611f5533bc787ab3bb45d0/Icon
#L0)):*
delete
___
*[msgs/Distance.proto, line 3 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAv5K49aOxLP-6QiYu:-MSPcGqMAD-bWjc1dzvs:b-8dahjt) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/msgs/Distance.proto#L3)):*
<details><summary><i>Previously, xabierolaz wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Still there
___
*[src/gazebo_distance_plugin.h, line 1 at r2](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSPcJcaCjGor4Z9pEDL:-MSPcJcaCjGor4Z9pEDM:b9xz1i1) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/1642ce295c91e7cb9b5df1f2c8a61ac34978e465/src/gazebo_distance_plugin.h#L1)):*
> ```objc
> /*
> ```

If this file isn't reusing anything from what it was copied from remove it. 


<!-- Sent from Reviewable.io -->

xabierolaz committed:
commit_fix

xabierolaz committed:
Delete Icon

xabierolaz committed:
Delete include/rotors_model directory

xabierolaz committed:
Delete common.h

xabierolaz committed:
Delete gazebo_imu_plugin.h

xabierolaz committed:
Delete gazebo_motor_model.h

xabierolaz committed:
Delete .DS_Store

xabierolaz committed:
Delete .gitignore

xabierolaz committed:
Delete Icon

xabierolaz committed:
Delete Icon

xabierolaz committed:
Delete gazebo_imu_plugin.cpp

xabierolaz committed:
Delete gazebo_imu_plugin.h

xabierolaz committed:
Delete gazebo_motor_model.cpp

xabierolaz committed:
Delete gazebo_motor_model.h

xabierolaz committed:
Delete tags

xabierolaz committed:
Delete CommandMotorSpeed.proto

xabierolaz committed:
Delete EscSensor.proto

xabierolaz committed:
Delete Icon

xabierolaz committed:
Unnecessary_files

xabierolaz committed:
last_commit

xabierolaz committed:
Delete .DS_Store

xabierolaz committed:
Update gazebo_distance_plugin.h

xabierolaz reviewed (commented):
> Reviewed 6 of 6 files at r3.
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 18 unresolved discussions (waiting on @wil3)

*[a discussion](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSPcY3q6VTBR11IBvau:-MSSCC_C34fPXpEe1NGi:b8kijik) (no related file):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

Below you've said you've made  changes but they haven't been made. I double checked some on your branch and they don't exist there either. Before responding they are done make sure the current revision actually reflects the change being made. 
</blockquote></details>

My bad, sorry,  there was conflict in the git pull so changes weren't showing up, I have revised it meticulously and its now as should be 

---
___
*[.DS_Store, line  at r2](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSPc0w8C7QERmnHQ0sp:-MSS8zMh7RntbGduouLn:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/1642ce295c91e7cb9b5df1f2c8a61ac34978e465/.DS_Store#L)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

what is this file? delete
</blockquote></details>

Done.
___
*[Icon
, line  at r2](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSPc3Pa1j4itGcSQzQ3:-MSS90055fG5qpw26Htk:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/60551f5eff70d68923611f5533bc787ab3bb45d0/Icon
#L)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

delete
</blockquote></details>

Done.
___
*[msgs/Distance.proto, line 3 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAv5K49aOxLP-6QiYu:-MSS9hyGCjHDasd2CgRg:b-886gsx) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/msgs/Distance.proto#L3)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

Still there
</blockquote></details>

Fixed
___
*[src/gazebo_distance_plugin.h, line 1 at r2](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSPcJcaCjGor4Z9pEDL:-MSSCBlSC8RMekzp5KHW:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/1642ce295c91e7cb9b5df1f2c8a61ac34978e465/src/gazebo_distance_plugin.h#L1)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

If this file isn't reusing anything from what it was copied from remove it. 
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

xabierolaz commented:
My bad @wil3 , sorry, there was conflict in the git pull so changes weren't showing up
I have revised it meticulously and its now as should be

wil3 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 9 unresolved discussions (waiting on @xabierolaz)

*[a discussion](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeGUzUDfiiag6hTsFC:-MSeGUzUDfiiag6hTsFD:b-jn99s5) (no related file):*
There still appears to be a lot of clean up todo. Please read through your code and remove any dead unused code. 

Please also use a code formatter, there are a number of style issues. Code should be formatted using clang-format-6.0, you can add plugins to your editor like this one so your code is consistent, https://github.com/google/vim-codefmt

---
___
*[.gitignore, line 0 at r3](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeDPGaAmcglJFe-Vk8:-MSeDPGaAmcglJFe-Vk9:b-np034q) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/e113717b8cd2d265a7712dbf0bc20b1f70ce4205/.gitignore#L0)):*
why was this removed? this is required
___
*[msgs/Distance.proto, line 8 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAvtdC1AQJNo5qUOWj:-MSeEj2p1GoCe2zcrqrV:bd7kqz1) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/msgs/Distance.proto#L8)):*
<details><summary><i>Previously, xabierolaz wrote…</i></summary><blockquote>

Understand the issue, but It actually wont compile without that line
</blockquote></details>

Thats because you use it in gazebo_distance_plugin.cpp, it should be removed from there too. 
___
*[src/gazebo_distance_plugin.h, line 37 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAw4_SCY-XSnaYRmYA:-MSeDZ9f-osVWk49WAJ9:b-pna44d) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.h#L37)):*
<details><summary><i>Previously, xabierolaz wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Looks like they just moved up a line (thats why the red dots indicate)
___
*[src/gazebo_distance_plugin.h, line 33 at r3](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeDifi9K6VvyRFfRs8:-MSeDifi9K6VvyRFfRs9:b-w1u5zt) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/e113717b8cd2d265a7712dbf0bc20b1f70ce4205/src/gazebo_distance_plugin.h#L33)):*
> ```objc
> 
>   DistanceParameters()
>       : 
> ```

Need to fix this formatting, not sure when these are all on separate lines, in general should follow Google C++ style guide
___
*[src/gazebo_distance_plugin.cpp, line 86 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxJx3-a_vFIeSXjLu:-MSeFkNsALEHaY-X91gt:bg7lhnm) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L86)):*
<details><summary><i>Previously, xabierolaz wrote…</i></summary><blockquote>

Did not find any issue regarding this in.cpp or .h , could you elaborate this ?
</blockquote></details>

Is max range unused? I can't find where it is used in the source
___
*[src/gazebo_distance_plugin.cpp, line 147 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAwWII54totUMOk31o:-MSeEL98-6xngDDyX-1l:bf91bbx) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L147)):*
<details><summary><i>Previously, xabierolaz wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Still there (thats what the dots are) most editors will highlight extra whitespace so you know if it exists. Unnecessary whitespaces shouldnt exist because it creates git thrashing.
___
*[src/gazebo_distance_plugin.cpp, line 160 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxj3oC_JLjhLfgxG3:-MSeEwqPAo2-W5Ofqgdp:b-ya3las) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L160)):*
> ```
> distance_message_.set_time_usec
> ```
what I mean is remove this line, its unneeded and needs to be removed from the distance message definition too
___
*[src/gazebo_distance_plugin.cpp, line 1 at r3](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeFAE_4AYCvzjW31bv:-MSeFAE_4AYCvzjW31bw:b-bt73uc) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/e113717b8cd2d265a7712dbf0bc20b1f70ce4205/src/gazebo_distance_plugin.cpp#L1)):*
> ```cpp
> /*
> ```

Remove this header too
___
*[src/gazebo_distance_plugin.cpp, line 71 at r3](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeGOCDBwTd1zMehq-W:-MSeGOCDBwTd1zMehq-X:bj5vhix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/e113717b8cd2d265a7712dbf0bc20b1f70ce4205/src/gazebo_distance_plugin.cpp#L71)):*
> ```cpp
>   node_handle_->Init(namespace_);
> 
>   if (_sdf->HasElement("linkName"))
> ```

I cant find any use for this either where is this used?


<!-- Sent from Reviewable.io -->

xabierolaz committed:
Create .gitignore

xabierolaz committed:
fix

xabierolaz commented:
Hi @wil3, all changed have been addressed and fixed.
As you stated, deleting that line in both files makes it compile succesfully.
Also, formated everything using Clang as you suggested and deleted whitespaces too.
cmake and make compiles without errors.
I will open a PR in the gymfc-master branch as we get finishing with the prerequisites

xabierolaz reviewed (commented):
> Reviewed 4 of 4 files at r4.
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 9 unresolved discussions (waiting on @wil3 and @xabierolaz)

*[a discussion](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeGUzUDfiiag6hTsFC:-MShFXzt7euOK7PPHWaX:b-896fix) (no related file):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

There still appears to be a lot of clean up todo. Please read through your code and remove any dead unused code. 

Please also use a code formatter, there are a number of style issues. Code should be formatted using clang-format-6.0, you can add plugins to your editor like this one so your code is consistent, https://github.com/google/vim-codefmt
</blockquote></details>

Done.

---
___
*[.gitignore, line  at r3](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeDPGaAmcglJFe-Vk8:-MSgTYT66SuMHSK38_XS:bgdmu8t) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/e113717b8cd2d265a7712dbf0bc20b1f70ce4205/.gitignore#L)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

why was this removed? this is required
</blockquote></details>

Done.Restored
___
*[msgs/Distance.proto, line 8 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAvtdC1AQJNo5qUOWj:-MShF_t04NqRQdgrNAu6:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/msgs/Distance.proto#L8)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

Thats because you use it in gazebo_distance_plugin.cpp, it should be removed from there too. 
</blockquote></details>

Done.
___
*[src/gazebo_distance_plugin.h, line 33 at r3](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeDifi9K6VvyRFfRs8:-MShFcLlA_AQ9UJstiMC:b-896fix) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/e113717b8cd2d265a7712dbf0bc20b1f70ce4205/src/gazebo_distance_plugin.h#L33)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

Need to fix this formatting, not sure when these are all on separate lines, in general should follow Google C++ style guide
</blockquote></details>

Done.
___
*[src/gazebo_distance_plugin.cpp, line 71 at r3](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeGOCDBwTd1zMehq-W:-MShJS3jEYXkKlfn5h3G:b-82rmqs) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/e113717b8cd2d265a7712dbf0bc20b1f70ce4205/src/gazebo_distance_plugin.cpp#L71)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

I cant find any use for this either where is this used?
</blockquote></details>

We have it in imu so I thought it is necessary for the sensor to be rested on some link so I have kept it there if that is not the case I can remove it


<!-- Sent from Reviewable.io -->

xabierolaz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 9 unresolved discussions (waiting on @wil3 and @xabierolaz)

___
*[src/gazebo_distance_plugin.cpp, line 86 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxJx3-a_vFIeSXjLu:-MShNPrbBazHO71U7ZTM:b-13bs0o) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L86)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

Is max range unused? I can't find where it is used in the source
</blockquote></details>

ok I have used it now I have just taken it for the comparison .


<!-- Sent from Reviewable.io -->

xabierolaz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 9 unresolved discussions (waiting on @wil3 and @xabierolaz)

___
*[src/gazebo_distance_plugin.cpp, line 147 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAwWII54totUMOk31o:-MShNo1n7FIRdCyPMQ_f:bnxm0mo) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L147)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

Still there (thats what the dots are) most editors will highlight extra whitespace so you know if it exists. Unnecessary whitespaces shouldnt exist because it creates git thrashing.
</blockquote></details>

Now I have used one tool form the vs code to remove the whitespace I hope that is solved


<!-- Sent from Reviewable.io -->

xabierolaz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 9 unresolved discussions (waiting on @wil3 and @xabierolaz)

___
*[src/gazebo_distance_plugin.cpp, line 160 at r1](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSAxj3oC_JLjhLfgxG3:-MShO1SC9_tsJtDnE5E-:bb6un1j) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/a6d65aae58b9518b737128f3395f6769c2b60bc1/src/gazebo_distance_plugin.cpp#L160)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

> ```
> distance_message_.set_time_usec
> ```
what I mean is remove this line, its unneeded and needs to be removed from the distance message definition too
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

xabierolaz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 9 unresolved discussions (waiting on @wil3)

___
*[src/gazebo_distance_plugin.cpp, line 1 at r3](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeFAE_4AYCvzjW31bv:-MShO6yU8x1MD3MplqxR:bb6un1j) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/e113717b8cd2d265a7712dbf0bc20b1f70ce4205/src/gazebo_distance_plugin.cpp#L1)):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

Remove this header too
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

wil3 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 8 unresolved discussions (waiting on @wil3 and @xabierolaz)

*[a discussion](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeGUzUDfiiag6hTsFC:-MT9OmrvDjSyfU6DfItC:bc61d95) (no related file):*
<details><summary><i>Previously, xabierolaz wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I'm still seeing unused code. Code should only exist if it has a purpose and you are using it. PRs should only be opened when the code is ready to be reviewed and merged, not when its a work in progress. I haven't even been able to review the logic yet because basic issues haven't been resolved. I've been patient because I'm guessing this is your first contribution to an open source project but in any other situation this PR would have already been rejected. We'll give this one more shot. 

---
___
*[src/gazebo_distance_plugin.h, line 17 at r4](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MT9Nk9J91MnNRaqZCYC:-MT9N5onBmagMocho2we:b-m2ulig) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/fb5a8de58f53cb29a2de0dcda93272fbfbf349ac/src/gazebo_distance_plugin.h#L17)):*
> ```objc
> 
> namespace ground_distance_units {
>     static const std::string METERS="m";
> ```


where is this used?
___
*[src/gazebo_distance_plugin.h, line 20 at r4](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MT9NMIK18DPcYeEYLT0:-MT9NMIK18DPcYeEYLT1:b-jmepyd) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/fb5a8de58f53cb29a2de0dcda93272fbfbf349ac/src/gazebo_distance_plugin.h#L20)):*
> ```objc
>     }
> namespace gazebo {
> //typedef const boost::shared_ptr<const sensor_msgs::msgs::Distance> Distanceptr;
> ```

unused
___
*[src/gazebo_distance_plugin.h, line 74 at r4](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MT9O6JMCyb4hfbKbsq2:-MT9O6JMCyb4hfbKbsq3:b2kz5vm) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/fb5a8de58f53cb29a2de0dcda93272fbfbf349ac/src/gazebo_distance_plugin.h#L74)):*
> ```objc
>   ignition::math::Temperature ground_distance_W_;
>   DistanceParameters distance_parameters_;
>   uint64_t seq_ = 0;
> ```

whats this for?
___
*[src/gazebo_distance_plugin.cpp, line 12 at r4](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MT9N_5aAcYivFiVxI8b:-MT9N_5aAcYivFiVxI8c:b-jmepyd) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/fb5a8de58f53cb29a2de0dcda93272fbfbf349ac/src/gazebo_distance_plugin.cpp#L12)):*
> ```cpp
> namespace gazebo
> {
>   /*  
> ```

unused
___
*[src/gazebo_distance_plugin.cpp, line 104 at r4](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MT9OOOR6ak9IDAObc3F:-MT9OOOS4hLdFPsX6yHZ:b-rohclz) ([raw file](https://github.com/wil3/gymfc-aircraft-plugins/blob/fb5a8de58f53cb29a2de0dcda93272fbfbf349ac/src/gazebo_distance_plugin.cpp#L104)):*
> ```cpp
>   ignition::math::Vector3d position = T_W_I.Pos();
>   float height = position.Z();
>   #if height <= MaxRange
> ```

not sure whats going on here, why is this a preprocessor directive?


<!-- Sent from Reviewable.io -->

xabierolaz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5)* status: all files reviewed, 8 unresolved discussions (waiting on @wil3 and @xabierolaz)

*[a discussion](https://reviewable.io/reviews/wil3/gymfc-aircraft-plugins/5#-MSeGUzUDfiiag6hTsFC:-MTesyGG0jQpWSW-bzCM:b-n72h9q) (no related file):*
<details><summary><i>Previously, wil3 (Wil Koch) wrote…</i></summary><blockquote>

I'm still seeing unused code. Code should only exist if it has a purpose and you are using it. PRs should only be opened when the code is ready to be reviewed and merged, not when its a work in progress. I haven't even been able to review the logic yet because basic issues haven't been resolved. I've been patient because I'm guessing this is your first contribution to an open source project but in any other situation this PR would have already been rejected. We'll give this one more shot. 
</blockquote></details>

You are absolutely right, I will treat this with the care and thoroughness expected in a PR as I never contributed to an open source project before. 
I ask you my sincerest apologies and I promise that my contributions from now on will be with the quality standards that are expected.

---


<!-- Sent from Reviewable.io -->
